### PR TITLE
Added formatting stuff

### PR DESCRIPTION
**Pull request by enncoded**
1. Added a github actions workflow to format future rules. Runs on PR and Push for both `develop` and `release`

### CODE REVIEW

File: .github/workflows/format-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "17.3.0"
+      - name: install prettier
+        run: npm install prettier
+      - name: format unformatted rules
+        run: npx prettier --write $(npx prettier --list-different "./**/*.yaml")
```
Comment (minusworld) [inline .github/workflows/format-rules.yaml:26]: Will this workflow file write changes back to the repository? I think that will require making a commit. Have you tested this on GitHub itself?

File: .github/workflows/num-rules.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: num-rules
```
Comment (minusworld) [inline .github/workflows/num-rules.yaml]: Do you know why this appears as "added?" I think this file was already in the repository. I guess maybe it got formatted?

Comment (minusworld) [reply .github/workflows/num-rules.yaml]: Oh, I see, it was renamed from `yml` to `yaml`.

**Comment by minusworld** [comment]
@lejliu I left a question about whether the workflow actually applies the autoformatting back to the repository.

Also, it might be beneficial to split this up into multiple PRs, two at minimum:
1. For the automatic formatting workflow
2. Reformatting everything according to the new formatter
- - You may also consider doing this in smaller batches. Maybe 2-3 PRs total. (Large amounts of changed files makes the GH UI choke on my machine/browser)

File: .github/workflows/format-rules.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      - name: commit formatted files
+        run: |
+          date > generated.txt
```
Comment (minusworld) [inline .github/workflows/format-rules.yaml]: What's the thinking behind this file?

Comment (enncoded) [reply .github/workflows/format-rules.yaml]: Oh that's a good point. That's not necessary. I'll remove that. I also noticed that the action fails when no files are changed, so I'll add a check for that too in my next push.

Comment (minusworld) [reply .github/workflows/format-rules.yaml]: 👍  Thanks!

**Comment by minusworld** [comment]
I wanted to see if we could test the workflow before committing and found https://github.com/nektos/act.

I made a few modifications based on how the tests went.

**Review by minusworld — APPROVED** [review]

**Pull request merged**